File: C/beginner/1181.c
#include <stdio.h>

int main() {
  
  double matrix[12][12];
  double total = 0;
  int L;
  char T;

  scanf("%d", &L);
  scanf("%c", &T);

  scanf("%c"); // cleaning Buffer

  for(int i = 0; i < 12; i++) {
    for(int j = 0; j < 12; j++) {
      scanf("%lf", &matrix[i][j]);
    }
  }

  for (int i = 0; i < 12; i++) total += matrix[L][i];
  
  if(T == 'M') total /= 12;
  printf("%.1lf\n", total);

  return 0;
}

File: dart/beginner/1181.dart
import 'dart:io';

void main() {
  var matrix = List.generate(12, (index) => List.filled(12, 0.0));
  double total = 0;
  int L = 0;
  String T = 'S';

  L = int.parse(stdin.readLineSync().toString().trim());
  T = stdin.readLineSync().toString().trim().toUpperCase();

  for (int i = 0; i < 12; i++) {
    for (int j = 0; j < 12; j++) {
      matrix[i][j] = double.parse(stdin.readLineSync().toString().trim());
    }
  }

  total = matrix[L].reduce((v, el) => v + el);

  if (T == 'M') total /= 12;

  print(total.toStringAsFixed(1));
}
